tests/tests/rpc_module.rs

```diff
@@ -340,12 +351,19 @@ async fn subscribing_without_server_bad_params() {
 async fn subscribe_unsubscribe_without_server() {
 	let mut module = RpcModule::new(());
 	module
-		.register_subscription("my_sub", "my_sub", "my_unsub", |_, mut sink, _| {
+		.register_subscription("my_sub", "my_sub", "my_unsub", |_, pending, _| {
 			let interval = interval(Duration::from_millis(200));
-			let stream = IntervalStream::new(interval).map(move |_| 1);
+			let mut stream = IntervalStream::new(interval).map(move |_| 1);
 
 			tokio::spawn(async move {
-				sink.pipe_from_stream(stream).await;
+				let sink = pending.accept().await.unwrap();
+
+				while let Some(item) = stream.next().await {
+					let msg = sink.build_message(&item).unwrap();
+					if sink.send(msg).await.is_ok() {
+						return;
+					}
+				}
```

Comment:
I guess whether we expose a `pipe_from_stream` or allow people to write this sort of loop manually, the end result is the same; if the sink can't consume items as quickly as the stream produces them, we'll end up with a buildup somewhere. (and if the stream happens to be bounded, then that backpressure from the slow sink will work its way all the way back and give the thing producing the stream inputs a chance to do what it needs to anyway).

All of that said, just a thought is to offer something that can function like:

```rust
// Just pipe the next item into the stream. If the last item is `Some(value)`, 
// we choose to not send it to the stream and just try sending the next (latest) value each time.
sink.pipe_from_stream_aggregating(|last: Option<Val>, next: Val| next, stream).await?;

// Until each item is consumed by the stream, we ignore next items coming in
sink.pipe_from_stream_aggregating(|last: Option<Val>, next: Val| last.unwrap_or(next), stream).await?;

// Run some function which combines the last (if present) and next value and queues that to be sent next.
sink.pipe_from_stream_aggregating(|last: Option<Val>, next: Val| aggregate(last, next), stream).await?;
```

Perhaps this would make it easy for people to use an aggregation approach when the sink is slow to consume things. (and then `pipe_from_stream_all` might exist to just buffer all messages up and allow the stream itself to do somethign with the backpressure if it is able, or else if the stream is an unbounded channel, we'll have a leak (we could document this).

Reply:
yupp, sounds like a good idea have this but I think we should kind of agree on the rest of the APIs before adding this back :)

Reply:
Yup that makes sense; just wanted to get it in writing somewhere while I had it in mind :)

Reply:
I'm still not sure whether we should use `send` or `try_send` in pipe_from_stream.

I did it with `try_send` because I don't want to "block" reading the stream,